### PR TITLE
Do not automatically close the client passed in

### PR DESCRIPTION
**Pull request by mcollina**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -8,7 +8,13 @@ const MongoClient = MongoDb.MongoClient
 const ObjectId = MongoDb.ObjectId
 
 function decorateFastifyInstance (fastify, client, options, next) {
-  fastify.addHook('onClose', (fastify, done) => client.close(done))
+  fastify.addHook('onClose', (fastify, done) => {
```
Comment (allevo) [inline index.js]: We can skip the hook registration at all.

Comment (mcollina) [reply index.js]: 👍 , I'll skip the hook then.

**Comment by mcollina** [comment]
> Why? This was a feature well documented.

The main reason is that you might want to pass a client so that it is shared by multiple things. If it's shared, Fastify might disrupt something else by closing it.
In the specific case, it prevents users to reuse the same db connections in unit tests: this increase the time to run them significantly.

**Comment by mcollina** [comment]
@allevo PTAL

**Pull request merged**